Numpy 2 release

Numpy 2 is released and is now being installed in our newer python environments.

There was a minor namespace change that created some CI failures.

```
      from numpy.core.numeric import ComplexWarning
  E   ImportError: cannot import name 'ComplexWarning' from 'numpy.core.numeric' (/home/runner/work/ASPIRE-Python/ASPIRE-Python/.tox/py3.10-pip/lib/python3.10/site-packages/numpy/core/numeric.py)
  ```
  
  I'll be pushing a minor patch momentarily.

--- a/src/aspire/numeric/complex_pca/validation.py
+++ b/src/aspire/numeric/complex_pca/validation.py
@@ -15,7 +15,14 @@ import warnings
 
 import numpy as np
 import scipy.sparse as sp
-from numpy.core.numeric import ComplexWarning
+from packaging.version import Version
+
+# Support both numpy 1 and 2
+if Version(np.__version__) >= Version("2.0.0"):
+    from numpy.exceptions import ComplexWarning
+else:
+    from numpy.core.numeric import ComplexWarning
+
 from sklearn.exceptions import DataConversionWarning
 from sklearn.utils.validation import _assert_all_finite
 

